modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -620,8 +619,12 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
         }
 
         if (currentTime > experiment.getEndTime().getTime()) {
+            // the experiment moves to the PAUSED state if the endtime is reached
+            experiment.setState(Experiment.State.PAUSED);
+            experimentUtil.updateExperimentState(experiment, Experiment.State.PAUSED);
```

Comment:
@AndreaSuckro - Every time the `experimentUtil.updateExperimentState` method is called, an email is sent out to the admin of the experiment with the transition details of the state obtained as `experiment.getState().toString()` corresponds to previous state and the second parameter in the method corresponds to the new state. Refer to the `updateExperimentState()` method implementation.

So, there are 2 things that need to be fixed here:

1) Experiment state should only be updated if the experiment is in `RUNNING` state already. That way we would guarantee that the experiment state change is attempted only once and only one email gets sent out. To ensure guarantees around that, please refer to the last section of the `isAssignmentValidForRapidExperiment` method in the same class. You must ensure that you call the update and cache refresh event only once, as far as possible, avoiding all race conditions.

2) While calling the `updateExperimentState` method, please keep the state of the `experiment` object unchanged as it is used to refer to the existing state of the experiment in the state transition email.